lib/ansible/modules/system/vdo.py

```diff
@@ -0,0 +1,762 @@
+              is on stable storage.  The 'async' mode acknowledges
+              writes when data has been cached for writing to stable
+              storage.
+        required: false
+    indexmem:
+        description:
+            - Specifies the amount of index memory in gigabytes.  The
+              default is 0.25.  The special decimal values 0.25, 0.5,
+              and 0.75 can be used, as can any positive integer.
+        required: false
+    indexmode:
+        description:
+            - Specifies the index mode of the Albireo index.  The
+              default is 'dense', which has a deduplication window of
+              1 GB of index memory per 1 TB of incoming data,
+              requiring 10 GB of index data on persistent storage.
+              The 'sparse' mode has a deduplication window of 1 GB of
+              index memory per 10 TB of incoming data, but requires
+              100 GB of index data on persistent storage.
+        required: false
+    ackthreads:
+        description:
+            - Specifies the number of threads to use for
+              acknowledging completion of requested VDO I/O operations.
+              Valid values are integer values from 1 to 100 (lower
+              numbers are preferable due to overhead).  The default is
+              1.
+        required: false
+    biothreads:
+        description:
+            - Specifies the number of threads to use for submitting I/O
+              operations to the storage device.  Valid values are
+              integer values from 1 to 100 (lower numbers are
+              preferable due to overhead).  The default is 4.
+        required: false
+    cputhreads:
+        description:
+            - Specifies the number of threads to use for CPU-intensive
+              work such as hashing or compression.  Valid values are
+              integer values from 1 to 100 (lower numbers are
+              preferable due to overhead).  The default is 2.
+        required: false
+    logicalthreads:
+        description:
+            - Specifies the number of threads across which to
+              subdivide parts of the VDO processing based on logical
+              block addresses.  Valid values are integer values from
+              1 to 100 (lower numbers are preferable due to overhead).
+              The default is 1.
+        required: false
+    physicalthreads:
+        description:
+            - Specifies the number of threads across which to
+              subdivide parts of the VDO processing based on physical
+              block addresses.  Valid values are integer values from
+              1 to 16 (lower numbers are preferable due to overhead).
+              The physical space used by the VDO volume must be
+              larger than (slabsize * physicalthreads).  The default
+              is 1.
+        required: false
+notes:
+  - In general, the default thread configuration should be used.
+'''
```

Comment:
I can add a "requirements:" section.

On RHEL 7, "PyYAML" is in the dependency list for the ansible RPM.  If you run "yum install ansible", it will try to also install PyYAML.  That aside, there is a "try:... except ImportError" wrapper around the "import yaml" call (which fixed the "ansible-test sanity --test import" test failure above).  This is exactly how a number of other Ansible 2.5 upstream modules handle ImportError exceptions.

Reply:
the module can run on systems in which ansible is not installed, that is why you should always put any 'non core' python library in requirements, even if you alert later on it is 'nicer' to inform the user when he reads the docs vs only when executing the plugin.